mempool/src/handle.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use crate::error::Error;
+use crate::tx_accumulator::TransactionAccumulator;
+use crate::MempoolEvent;
+use crate::MempoolInterface;
+use common::chain::signed_transaction::SignedTransaction;
+use common::chain::Transaction;
+use common::primitives::Id;
+use tokio::sync::mpsc;
+use tokio::sync::oneshot;
+
+pub struct MempoolInterfaceHandle {
+    sender: mpsc::UnboundedSender<MempoolMethodCall>,
+}
+
+impl MempoolInterfaceHandle {
+    pub fn new(sender: mpsc::UnboundedSender<MempoolMethodCall>) -> Self {
+        Self { sender }
+    }
+}
+
+pub enum MempoolMethodCall {
```

Comment:
This shouldn't be public. It glosses on the hierarchical design issue. Things that need this should go under it and shouldn't be exposed to this through public. In other words: Why would `lib.rs` need anything to do with this?

Reply:
I moved this to its own submodule of mempool_interface_impl so that its public but no longer accessible to `lib.rs`. Does that make sense?